iceoryx_posh/include/iceoryx_posh/internal/popo/wait_set.inl

```diff
@@ -19,13 +19,129 @@ namespace iox
 {
 namespace popo
 {
+template <uint64_t Capacity>
+inline WaitSet<Capacity>::WaitSet() noexcept
+    : WaitSet(runtime::PoshRuntime::getInstance().getMiddlewareConditionVariable())
+{
+}
+
+template <uint64_t Capacity>
+inline WaitSet<Capacity>::WaitSet(cxx::not_null<ConditionVariableData* const> condVarDataPtr) noexcept
+    : m_conditionVariableDataPtr(condVarDataPtr)
+    , m_conditionVariableWaiter(m_conditionVariableDataPtr)
+{
+}
+
+template <uint64_t Capacity>
+inline WaitSet<Capacity>::~WaitSet() noexcept
+{
+    removeAllTriggers();
+    /// @todo Notify RouDi that the condition variable data shall be destroyed
+}
+
+template <uint64_t Capacity>
+inline void WaitSet<Capacity>::removeTrigger(const uint64_t uniqueTriggerId) noexcept
+{
+    for (auto& currentTrigger : m_triggerVector)
+    {
+        if (currentTrigger.getUniqueId() == uniqueTriggerId)
+        {
+            currentTrigger.invalidate();
```

Comment:
I guess this invalidated and not reset like in `removeAllTriggers` because there might be an active user?

Reply:
The reset call is used to remove the trigger in the counterpart and if you call remove the counterpart has to call invalidate